inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/utils/SharePreferencesUtil.kt

```diff
@@ -1,17 +1,79 @@
 package com.rakuten.tech.mobile.inappmessaging.runtime.utils
 
 import android.content.Context
+import android.content.SharedPreferences
 import androidx.security.crypto.EncryptedSharedPreferences
 import androidx.security.crypto.MasterKey
+import timber.log.Timber
+import java.io.File
+import java.security.KeyStore
 
 internal object SharePreferencesUtil {
-    fun generateKey(context: Context) = MasterKey.Builder(context).setKeyScheme(MasterKey.KeyScheme.AES256_GCM).build()
-
-    fun createSharedPreference(context: Context, key: MasterKey, account: String) = EncryptedSharedPreferences.create(
-                context,
-                account,
-                key,
-                EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
-                EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
-        )
+
+    @SuppressWarnings("TooGenericExceptionCaught", "ReturnCount", "LongMethod", "NestedBlockDepth")
+    fun createSharedPreference(
+        context: Context,
+        account: String,
+        key: MasterKey? = null,
+        shouldRetry: Boolean = true,
+        timber: Timber.Tree = Timber.tag(TAG)
+    ): SharedPreferences {
+        if (isEncryptedEnabled(context)) {
```

Comment:
`isEncryptedEnabled` returns true for freshly installed apps?

Reply:
yes, it will return true since the default value is set to `true`

Reply:
Ah ok, I can see it now :)